### PR TITLE
add sales_team_multicompany module

### PR DESCRIPTION
**Pull request by chafique-delli**

### CODE REVIEW

File: sales_team_multicompany/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Sales Team Multicompany',
+    'summary': 'Add companies in sales team and '
+               'filter the sales team from the company',
+    'version': '9.0.0.1.0',
```
Comment (pedrobaeza) [inline sales_team_multicompany/__openerp__.py]: 9.0.1.0.0

File: sales_team_multicompany/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'category': 'Sales Management',
+    'website': 'http://akretion.com',
+    'author': 'Akretion',
+    'license': 'AGPL-3',
+    'application': False,
+    'installable': True,
+    'external_dependencies': {
```
Comment (pedrobaeza) [inline sales_team_multicompany/__openerp__.py]: Don't put empty keys

File: sales_team_multicompany/demo/res.users.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+id,name,login,groups_id/id,company_id/id,company_ids/id
+res_users1,User company 1,user_company1@gmail.com,"base.group_user,base.group_sale_salesman",res_company1,res_company1
```
Comment (pedrobaeza) [inline sales_team_multicompany/demo/res.users.csv]: Why don't you use existing users?

File: sales_team_multicompany/sales_team.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+from openerp import models, fields
+
+
+class CrmTeam(models.Model):
+    _inherit = 'crm.team'
+
+    company_ids = fields.Many2many(
+        'res.company',
+        string='Companies',
+        required=True,
```
Comment (pedrobaeza) [inline sales_team_multicompany/sales_team.py]: Don't make it required. The standard interpretation is empty for allowing all companies

File: sales_team_multicompany/security/ir.rule.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,groups:id,perm_read,perm_write,perm_create,perm_unlink,domain_force
+sales_team.crm_team_rule,crm.team multi-company,sales_team.model_crm_team,,True,False,False,False,"['|', ('company_ids', '=', 'user.company_id.id'), ('company_ids','child_of',[user.company_id.id])]"
```
Comment (pedrobaeza) [inline sales_team_multicompany/security/ir.rule.csv]: The rules are usually put in XML and with a noupdate="1" attribute for allowing to change the rule manually.

Comment (chafique-delli) [inline sales_team_multicompany/security/ir.rule.csv]: @pedrobaeza, Thanks for your remarks, I corrected.

File: sales_team_multicompany/README.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=========================
+Sales Team Multi-Company
```
Comment (elicoidal) [inline sales_team_multicompany/README.rst]: Multi-company Sales Team

File: sales_team_multicompany/README.rst [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+sales_team_multicompany%0Aversion:%20
+8.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Chafique Delli (chafique.delli@akretion.com)
```
Comment (elicoidal) [inline sales_team_multicompany/README.rst]: use standard email formatting: Chafique Delli chafique.delli@akretion.com

File: sales_team_multicompany/README.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (elicoidal) [inline sales_team_multicompany/README.rst]: https

File: sales_team_multicompany/demo/res.company.csv [hunk 9]
```diff
@@ -0,0 +1,3 @@
+id,name,currency_id/id
+res_company1,Company 1,base.EUR
```
Comment (elicoidal) [inline sales_team_multicompany/demo/res.company.csv]: why dont use existing demo companies?

File: sales_team_multicompany/sales_team.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+from openerp import models, fields
+
+
+class CrmTeam(models.Model):
+    _inherit = 'crm.team'
+
+    company_ids = fields.Many2many(
+        'res.company',
+        string='Companies',
+        help="Add the companies concerned by this market.")
```
Comment (elicoidal) [inline sales_team_multicompany/sales_team.py]: Company to which the sales team belongs

**Comment by elicoidal** [comment]
Thanks for the contribution. Demo data should be standard ones. Other than small details, LGTM

**Comment by chafique-delli** [comment]
@elicoidal, Thanks for your remarks.
@elicoidal,@pedrobaeza, Regarding the demo data, I have noted your 
suggestions that I will apply in the future, but my demo data do not 
alter my tests that are 100% reliable and does not deserve a lock on 
this P.R.

**Comment by pedrobaeza** [comment]
I think you should adapt your demo data and test now. It's very simple.

**Comment by bealdav** [comment]
@pedrobaeza "the best is the enemy of the good'. Please consider than demo here are completely valid for tests purpose. @akretion we move towards unit tests for our new modules, it's a step in the right direction. We should encourage our developers to contribute.
We are not perfect, let us progressing and go ahead ...
Thanks for your huge work on reviews from all odoo community

**Comment by pedrobaeza** [comment]
I know, but the same I invested some time in your suggestions made for base_iso3166 and others modules reviewed by you, I ask the same for this one, for not having a lot of demo data (companies, users...) that are only valid for one module, and that will interfere on manual tests on runbot, confusing the user.

**Comment by bealdav** [comment]
@pedrobaeza from a contributor like you, I can only accept these remarks, it will reduce the confusion.
Thanks

**Comment by chafique-delli** [comment]
@pedrobaeza, I changed the demo data, can you tell me if it's ok now .
Thanks,

File: sales_team_multicompany/sales_team_multicompany_demo.xml [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
```
Comment (pedrobaeza) [inline sales_team_multicompany/sales_team_multicompany_demo.xml]: Is this data originally `noupdate="1"`. If so, it won't be updated with this definition. You need to set a pre and post-init-hook to remove the noupdate flag and after the installation, to restore it.

You can find an example here: https://github.com/OCA/l10n-spain/pull/214/files#diff-d1d1fa00d15ff44bb400415153c94f00R30 (but you have to use latest conventions that move these methods to the file hooks.py).

**Comment by pedrobaeza** [comment]
Thanks for the efforts

:+1:

**Comment by chafique-delli** [comment]
@pedrobaeza, Thanks for your reviews.

**Comment by sebastienbeau** [comment]
LGTM :+1: code review

**Comment by mourad-ehm** [comment]
:+1:  code review

**Pull request merged**